File: BinarytoDecimal.kt
package BinaryDecimal

import kotlin.math.pow

fun binaryDecimal(binary_number: Long): Int {
	val value: String = binary_number.toString()
	val len: Int = value.length
	var output: Int = 0
	var char: Char
	for (position in value.indices) {
		char = value[position]
		if (char == '1') {
			output += 2.0.pow(len - 1.0 - position).toInt()
		} else if (char == '0') {
			continue
		} else {
			System.err.println("Make sure you only input binary values")
			return -1
		}
	}
	return output
}

fun main() {
	print("Binary Number: ")
	println(binaryDecimal(Integer.valueOf(readLine()).toLong()))
}

File: BinarytoDecimal.exs
#!/usr/bin/env elixir

defmodule BinarytoDecimal do
    def binary_to_decimal(binary_number) do
        value = Integer.to_string(binary_number)
        len = String.length(value)
        Enum.sum(for position <- 0..len-1 do
            char = String.at(value, position)
            cond do
                char == "1" ->
                    :math.pow(2, (len - 1 - position)) |> round
                char == "0" ->
                    0
                true ->
                    IO.puts("Make sure you only input binary values")
                    exit(-1)
            end
        end)
    end
end

IO.puts(BinarytoDecimal.binary_to_decimal(String.to_integer(String.trim(IO.gets("Binary Number: ")))))

